main/src/main/java/cgeo/geocaching/unifiedmap/layers/ElevationChartUtils.java

```diff
@@ -0,0 +1,169 @@
+package cgeo.geocaching.unifiedmap.layers;
+
+import cgeo.geocaching.CgeoApplication;
+import cgeo.geocaching.R;
+import cgeo.geocaching.location.Geopoint;
+import cgeo.geocaching.location.Units;
+import cgeo.geocaching.models.Route;
+import cgeo.geocaching.models.RouteSegment;
+import cgeo.geocaching.models.geoitem.GeoIcon;
+import cgeo.geocaching.models.geoitem.GeoItem;
+import cgeo.geocaching.models.geoitem.GeoPrimitive;
+import cgeo.geocaching.unifiedmap.geoitemlayer.GeoItemLayer;
+import cgeo.geocaching.utils.ImageUtils;
+import static cgeo.geocaching.unifiedmap.LayerHelper.ZINDEX_ELEVATIONCHARTMARKERPOSITION;
+import static cgeo.geocaching.utils.DisplayUtils.getDimensionInDp;
+
+import android.content.res.Resources;
+import android.view.View;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.core.content.res.ResourcesCompat;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import com.github.mikephil.charting.charts.LineChart;
+import com.github.mikephil.charting.components.Legend;
+import com.github.mikephil.charting.components.XAxis;
+import com.github.mikephil.charting.components.YAxis;
+import com.github.mikephil.charting.data.Entry;
+import com.github.mikephil.charting.data.LineData;
+import com.github.mikephil.charting.data.LineDataSet;
+import com.github.mikephil.charting.highlight.Highlight;
+import com.github.mikephil.charting.listener.OnChartValueSelectedListener;
+
+public class ElevationChartUtils {
+
+    private static final String ELEVATIONCHART_MARKER = "ELEVATIONCHARTMARKER";
+
+    private ElevationChartUtils() {
+        // utility class
+    }
+
+    public static void toggleElevationChart(@NonNull final AppCompatActivity activity, final Route route, final GeoItemLayer<String> geoItemLayer, final String title) {
+        final View chartBlock = activity.findViewById(R.id.elevation_block);
+        final LineChart chart = activity.findViewById(R.id.elevation_chart);
+        if (chart == null) {
+            return;
+        }
+        if (chartBlock.getVisibility() == View.VISIBLE || route == null) {
+            chartBlock.setVisibility(View.GONE);
```

Comment:
Just discovered another bug, you should also remove the `ELEVATIONCHART_MARKER` when hiding the elevation chart

Reply:
done